modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenPasswordGrantTestCase.java

```diff
@@ -0,0 +1,403 @@
+        authorizeDomainAPIs(application.getId(), domainAPIId, customScopes);
+
+        addUserWithRole(application.getId(), customScopes);
+
+        MockServer.createMockServer(MOCK_SERVER_ENDPOINT);
+        createPreIssueAccessTokenAction();
+
+        accessToken = retrieveAccessToken(application.getId(), customScopes);
+        jwtClaims = extractJwtClaims(accessToken);
+    }
+
+    @Test(groups = "wso2.is", description = "Verify the presence of the added scope in the access token")
+    public void testTokenScopeOperations() throws Exception {
+
+        String[] scopes = jwtClaims.getStringClaim("scope").split("\\s+");
+
+        Assert.assertTrue(ArrayUtils.contains(scopes, NEW_SCOPE_1));
+        Assert.assertTrue(ArrayUtils.contains(scopes, NEW_SCOPE_4));
+        Assert.assertFalse(ArrayUtils.contains(scopes, CUSTOM_SCOPE_3));
+        Assert.assertFalse(ArrayUtils.contains(scopes, CUSTOM_SCOPE_2));
+    }
+
+    @Test(groups = "wso2.is", description = "Verify the presence of the added scope in the access token")
+    public void testTokenAUDClaimOperations() throws Exception {
+
+        String[] audValueArray = jwtClaims.getStringArrayClaim("aud");
+
+        Assert.assertTrue(ArrayUtils.contains(audValueArray, "zzz1.com"));
+        Assert.assertTrue(ArrayUtils.contains(audValueArray, "zzzR.com"));
+        Assert.assertFalse(ArrayUtils.contains(audValueArray, clientId));
+    }
+
+    @DataProvider(name = "getStringClaim")
+    public Object[][] getStringClaim() {
+
+        return new Object[][]{
+                {"custom_claim_string_1", "testCustomClaim1", 0},
+                {"custom_claim_string_2", "testCustomClaim2", 1},
+                {"custom_claim_string_3", "testCustomClaim3", 2}
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Verify the presence of the specified custom string claim in the access " +
+            "token", dataProvider = "getStringClaim")
+    public void testTokenStringClaimAddOperation(String claim, String newClaimValue, int index) throws Exception {
+
+        String claimStr = jwtClaims.getStringClaim(claim);
+        if (index == 0) {
+            Assert.assertEquals(claimStr, newClaimValue);
+        }
+    }
+
+    @DataProvider(name = "getNumberClaim")
+    public Object[][] getNumberClaim() {
+
+        return new Object[][]{
+                {"custom_claim_number_1", 78, 0},
+                {"custom_claim_number_2", 50, 1},
+                {"custom_claim_number_3", 32, 2},
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Verify the presence of the specified custom number claim in the access " +
+            "token", dataProvider = "getNumberClaim")
+    public void testTokenNumberClaimAddOperation(String claim, Number newClaimValue, int index) throws Exception {
+
+        Number claimValue = jwtClaims.getIntegerClaim(claim);
+        if (index == 0) {
+            Assert.assertEquals(claimValue, newClaimValue);
+        }
+    }
```

Comment:
Refer the comment above. Have the same concerns with regard to this method

Reply:
Updated with 4412a6451aab9f4e6c947538138d3c80ad3f36b7